src/test-utils/dom/file-upload/index.ts

```diff
@@ -0,0 +1,84 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { ComponentWrapper, ElementWrapper } from '@cloudscape-design/test-utils-core/dom';
+import fileUploadSelectors from '../../../file-upload/styles.selectors.js';
+import fileUploadInputSelectors from '../../../file-upload/file-input/styles.selectors.js';
+import fileUploadOptionSelectors from '../../../file-upload/file-option/styles.selectors.js';
+import tokenGroupSelectors from '../../../token-group/styles.selectors.js';
+import tokenListSelectors from '../../../internal/components/token-list/styles.selectors.js';
+import formFieldStyles from '../../../form-field/styles.selectors.js';
+import ButtonWrapper from '../button';
+
+export default class FileUploadWrapper extends ComponentWrapper<HTMLElement> {
+  static rootSelector: string = fileUploadSelectors.root;
+
+  findUploadButton(): ButtonWrapper {
+    return this.findComponent(`.${fileUploadInputSelectors['upload-button']}`, ButtonWrapper)!;
+  }
+
+  findNativeInput(): ElementWrapper<HTMLInputElement> {
+    return this.findByClassName(fileUploadInputSelectors['upload-input'])!;
+  }
+
+  findFileTokens(): Array<FileTokenWrapper> {
+    return this.findAllByClassName(tokenGroupSelectors.token).map(
+      tokenElement => new FileTokenWrapper(tokenElement.getElement())
+    );
+  }
+
+  /**
+   * Returns a file token from for a given index.
+   *
+   * @param tokenIndex 1-based index of the file token to return.
+   */
+  findFileToken(fileTokenIndex: number): null | FileTokenWrapper {
+    if (fileTokenIndex === 1) {
+      return this.findComponent(`.${tokenGroupSelectors.token}`, FileTokenWrapper);
+    }
+    return this.findComponent(
+      `.${tokenListSelectors['list-item']}:nth-child(${fileTokenIndex}) .${tokenGroupSelectors.token}`,
+      FileTokenWrapper
+    );
```

Comment:
What is happening here, why is there a conditon?

Reply:
When multiple=false at most one token is rendered and we do not use the token-list. When multiple=false we need to use token-list styles to locate the token by index.

Reply:
This implementation is based on fragile assumption then when there are multiple elements are found, it quietly clicks the first.

For example, this line returns multiple matches: 

https://github.com/cloudscape-design/components/blob/a6cc465c20ae4491e438a087c71f6bf80fe8c13f/src/file-upload/__integ__/file-upload.test.ts#L23

It works in Webdriver, but will fail in Cypress: 
https://docs.cypress.io/api/commands/click#Click-all-elements-with-id-starting-with-btn